providers/azure/provider/provider.go

```diff
@@ -134,6 +134,10 @@ func (s *Service) Connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 		return nil, err
 	}
 
+	for _, a := range inventory.GetSpec().GetAssets() {
```

Comment:
I am not 100 sure I understand this, wouldn't this be required to be pulled from the resource itself?

Reply:
The use case  I am trying to solve is passing in custom labels on the root that are not necessarily computable on each resource level.

An example is 'mondoo.com/integration-mrn' which can only be passed it, it cannot be determined by each resource

Reply:
ohh, bc when we connect the labels are added to the req.Asset and then all the discovered ones don't have the label. 

this made me think about annotations, e.g. `cnspec scan local --annotation test=vj` -- looks like we're not passing the annotations through either. ill open an issue

Reply:
I think there are two valid use cases:
1. Mondoo specific labels that should be propagated. I agree that we should do that outside of the providers itself, but then the copy needs to be restricted to specific labels that start with mondoo.com
2. I think it makes sense to support annotations and they can be copied over since they are user provided. We can do that also outside of the providers.

Reply:
I have changed to only take `mondoo.com/` labels into account and have moved the logic into the scanning flow itself